src/components/itemidentifier/itemidentifier.js

```diff
@@ -306,9 +304,12 @@ define(["dialogHelper", "loading", "connectionManager", "require", "globalize",
 
                 html += '<div class="inputContainer">';
 
-                var idLabel = globalize.translate("LabelDynamicExternalId").replace("{0}", idInfo.Name);
+                var fullName = idInfo.Name;
+                if (idInfo.Type) {
+                    fullName = idInfo.Name + " " + globalize.translate(idInfo.Type);
```

Comment:
```suggestion
                    fullName = idInfo.Name + " " + globalize.translate("MediaType:" + idInfo.Type);
```
to make it easier to understand the context for translating

Reply:
I definitely don't want this because it makes analyzing the strings currently in use a real headache. All translations should be static references.

Reply:
I don't see how it does. You cannot statically figure what comes in `idInfo.Type` via JSON anyway.

Reply:
Is this an existing pattern (scoping string resources to their particular use with a prefix)? I figured make them as generic as possible for reuse.

Reply:
> Is this an existing pattern

Not at the moment.

> You cannot statically figure what comes in idInfo.Type via JSON anyway.

You can with a switch statement, although that's almost as annoying. We really need to reduce the amount of translations in the web client though, and dynamic strings have been a huge pain so far, which is why I don't want to add any more.

Reply:
> I figured make them as generic as possible for reuse.

That's definitely what we want. We could probably get rid of over 200 strings right now if we just removed duplicates and made the keys more generic.

Reply:
Once we move over to something like i18next for translations, we can use a better format for translations (like .po files) and have context in the source files.